File: climbing-stairs/climbing-stairs.cs
public class Solution {
    Dictionary<int,int> map=new Dictionary<int,int>();
    
    public int ClimbStairs(int n) {
        if(n==0 || n==1)
        {
            return 1;
        }
        
        if(map.ContainsKey(n))
        {
            return map[n];
        }
        else
        {
            map.Add(n,ClimbStairs(n-1)+ClimbStairs(n-2));
        }
        
        return map[n];
    }
}

File: climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        
        int[] dp=new int[n];
        
        if(n==1 || n==2)
        {
            return n;
        }
            
        
        dp[0]=1; //1 step
        dp[1]=2; //2nd step
        
        for(int i=2;i<n;i++)
        {
            dp[i]=dp[i-2]+dp[i-1];
        }
        
        return dp[n-1];
    }
}